### ISSUE TITLE
Floating window sliding towards bottom right

### ISSUE THREAD

**Issue by Soupstraw** [xmonad/xmonad#423]
### Problem Description

When I have window borders enabled and run a program that starts in a floating window, the window starts sliding diagonally in the bottom-right direction.

### Steps to Reproduce

 * Load the configuration file provided below
 * Launch some program that launches in a floating window (e.g. steam login prompt)
 * Watch as the window escapes out the bottom-right corner of the screen

### Configuration File

```haskell

import XMonad hiding (float)
import XMonad.Config.Desktop
import XMonad.StackSet

myConfig = def
  { borderWidth = 3
  }


main :: IO ()
main = xmonad myConfig
```

### Checklist

  - [x] I've read [CONTRIBUTING.md](https://github.com/xmonad/xmonad/blob/master/CONTRIBUTING.md)

  - I tested my configuration
    - [x] With `xmonad` version 0.17.1
    - [x] With `xmonad-contrib` version 0.17.1

**Comment by liskin** [comment]
Oh, the joy of broken software that expects a reparenting WM… This should be reported to Steam.

I've had a workaround for Steam insisting on moving its floats in my xmonad.hs for a while which I've been meaning to upstream into xmonad-contrib but can't find the time to do it unfortunately:
https://github.com/liskin/dotfiles/blob/743c499907eb3fa031e6a9eb2f94f191495a9ac6/.xmonad/xmonad.hs#L300-L327
https://github.com/liskin/dotfiles/blob/743c499907eb3fa031e6a9eb2f94f191495a9ac6/.xmonad/xmonad.hs#L273-L278
https://github.com/liskin/dotfiles/blob/743c499907eb3fa031e6a9eb2f94f191495a9ac6/.xmonad/xmonad.hs#L292

Also, it's a massive hack compensating for the absence of a hook in https://github.com/xmonad/xmonad/blob/3d8238b35d0f27dcd9b23c58ea33b5772ec909e9/src/XMonad/Main.hs#L388-L398

**Comment by Soupstraw** [comment]
Yeah, I think I'll stick to leaving the border width to the default value, which seems to work around this issue for the time being.
